### PR TITLE
Improve TextSpan and For statements

### PR DESCRIPTION
**Pull request by sebastienros**
Extracted from the compilation branch

### CODE REVIEW

File: Fluid/Ast/ForStatement.cs [hunk 1]
```diff
@@ -36,6 +35,7 @@ public sealed class ForStatement : TagStatement
         public Expression Offset { get; }
         public bool Reversed { get; }
         public ElseStatement Else { get; }
+        public bool OffSetIsContinue { get; }
```
Comment (hishamco) [inline Fluid/Ast/ForStatement.cs:38]:
```suggestion
        public bool OffsetIsContinue { get; }
```

File: Fluid/Ast/ForStatement.cs [hunk 2]
```diff
@@ -25,7 +24,7 @@ public sealed class ForStatement : TagStatement
             Reversed = reversed;
             Else = elseStatement;
 
-            _isContinueOffset = Offset is MemberExpression l && l.Segments.Count == 1 && ((IdentifierSegment)l.Segments[0]).Identifier == "continue";
+            OffSetIsContinue = Offset is MemberExpression l && l.Segments.Count == 1 && ((IdentifierSegment)l.Segments[0]).Identifier == "continue";
```
Comment (hishamco) [inline Fluid/Ast/ForStatement.cs:27]:
```suggestion
            OffsetIsContinue = Offset is MemberExpression l && l.Segments.Count == 1 && ((IdentifierSegment)l.Segments[0]).Identifier == "continue";
```

File: Fluid/Ast/ForStatement.cs [hunk 3]
```diff
@@ -55,7 +55,7 @@ public override async ValueTask<Completion> WriteToAsync(TextWriter writer, Text
             var startIndex = 0;
             if (Offset is not null)
             {
-                if (_isContinueOffset)
+                if (OffSetIsContinue)
```
Comment (hishamco) [inline Fluid/Ast/ForStatement.cs:58]:
```suggestion
                if (OffsetIsContinue)
```

File: Fluid/Ast/ForStatement.cs (hunk 2 above)
Comment (hishamco) [inline Fluid/Ast/ForStatement.cs:27]: Can we used `static` field for "Continue" identifier

File: Fluid/Ast/TextSpanStatement.cs [hunk 4]
```diff
@@ -1,16 +1,15 @@
...
 {
     public sealed class TextSpanStatement : Statement
     {
-        private bool _isStripped;
-        private bool _isEmpty;
+        private bool _isBufferPrepared;
         private readonly object _synLock = new();
         private TextSpan _text;
-        private string _buffer;
+        internal string _preparedBuffer;
```
Comment (hishamco) [inline Fluid/Ast/TextSpanStatement.cs:12]: Why `internal`?

**Review by hishamco — CHANGES_REQUESTED** [review]

File: Fluid/Ast/TextSpanStatement.cs (hunk 4 above)
Comment (sebastienros) [reply Fluid/Ast/TextSpanStatement.cs:12]: It's used internally in the branch it's coming from, will keep it this way for now.

File: Fluid/Ast/ForStatement.cs (hunk 2 above)
Comment (sebastienros) [reply Fluid/Ast/ForStatement.cs]: If it was used somewhere else it could be a const, but it just local so it doesn't matter.

**Pull request merged**